### PR TITLE
Feature/random word selector

### PR DESCRIPTION
**Pull request by hothanhtuan1209**
In this pull request, i refactor random_word_selector.py file in practice-2, update function main and add module random_word_selector to main.py
#46

### CODE REVIEW

**Comment by hothanhtuan1209** [comment]
image

File: python/practices/practice-3/src/path2/random_word_selector.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    """
+
+    hist = collections.Counter()
+
+    for line in file_content:
+        line = line.replace('-', ' ')
+
+        for word in line.split():
+            word = word.strip(string.punctuation + string.whitespace)
+            word = word.lower()
```
Comment (cuongdoan-agilityio) [inline python/practices/practice-3/src/path2/random_word_selector.py]: Can we be more compact?
Please refer to: Writing Idiomatic Python 3 - Chapter 2: Working with Data - String

File: python/practices/practice-3/src/path2/random_word_selector.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            hist[word] = hist.get(word, 0) + 1
+
+    return hist
+
+
+def calculate_cumulative_sums(list_word):
```
Comment (cuongdoan-agilityio) [inline python/practices/practice-3/src/path2/random_word_selector.py]: Please update the parameter name.

File: python/practices/practice-3/src/path2/random_word_selector.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    Args:
+        hist (dict): A histogram of word frequencies.
+
+    Returns:
+        str: A randomly selected word from the histogram.
+    """
+
+    list_word = list(hist.keys())
+    calculate_sum = calculate_cumulative_sums(
```
Comment (cuongdoan-agilityio) [inline python/practices/practice-3/src/path2/random_word_selector.py]: Please update the variable name.

File: python/practices/practice-3/src/path2/random_word_selector.py (hunk 1 above)
Comment (hothanhtuan1209) [reply python/practices/practice-3/src/path2/random_word_selector.py]: Done

File: python/practices/practice-3/src/path2/random_word_selector.py (hunk 2 above)
Comment (hothanhtuan1209) [reply python/practices/practice-3/src/path2/random_word_selector.py]: Done

File: python/practices/practice-3/src/path2/random_word_selector.py (hunk 3 above)
Comment (hothanhtuan1209) [reply python/practices/practice-3/src/path2/random_word_selector.py]: Done

File: python/practices/practice-3/src/path2/random_word_selector.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import string
+import collections
+import os
+from random import randint
+from bisect import bisect
+from constants.constants import EBOOK
+from helpers.file_reader import read_file
+
+
+def clean_file(file_content):
```
Comment (cuongdoan-agilityio) [inline python/practices/practice-3/src/path2/random_word_selector.py]: Please update the name of the function and param.

File: python/practices/practice-3/src/path2/random_word_selector.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        list(int(i) for i in hist.values())
+    )
+    total = cumulative_sums[-1]
+
+    if total > 0:
+        random_val = randint(0, total - 1)
+        index = bisect(cumulative_sums, random_val)
+        return list_word[index]
+    else:
+        return "Histogram is empty."
```
Comment (cuongdoan-agilityio) [inline python/practices/practice-3/src/path2/random_word_selector.py]: I think it should return an empty string.

Comment (cuongdoan-agilityio) [inline python/practices/practice-3/src/path2/random_word_selector.py]: Please refer to Writing Idiomatic Python 3 to refactor this code.

File: python/practices/practice-3/src/path2/random_word_selector.py (hunk 4 above)
Comment (hothanhtuan1209) [reply python/practices/practice-3/src/path2/random_word_selector.py]: Done

File: python/practices/practice-3/src/path2/random_word_selector.py (hunk 5 above)
Comment (hothanhtuan1209) [reply python/practices/practice-3/src/path2/random_word_selector.py]: Done

Comment (hothanhtuan1209) [reply python/practices/practice-3/src/path2/random_word_selector.py]: Done

**Pull request merged**